src/ast/stmt.rs

```diff
@@ -15,70 +15,89 @@ pub enum Stmt {
     DoWhileStmt(DoWhileStmt),
     ReturnStmt(ReturnStmt),
     SwitchStmt(SwitchStmt),
-    IncDecStmt(IncDecStmt),
     ImportStmt(ImportStmt),
+    BreakStmt(BreakStmt),
+    ContinueStmt(ContinueStmt),
 }
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, new)]
 pub struct AssignStmt {
     // lhs could be something like *var
     pub lhs: Expr,
     pub rhs: Expr,
+    #[new(value = r#""assign_stmt""#)]
+    r#type: &'static str,
 }
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, new)]
 pub struct DeclStmt {
-    pub lhs: Ident,
-    pub rhs: Vec<Expr>, // Vec since FieldComponent could declare multiple variables
+    /// The type of the declared variable(s).
+    pub var_type: Option<String>,
+    /// The variable(s). These could be Idents (x, y, z), BinExprs (x = y = 32) etc.
+    pub rhs: Vec<Expr>,
+    #[new(default)]
+    pub is_static: Option<bool>,
+    #[new(default)]
+    pub is_final: Option<bool>,
+    #[new(value = r#""decl_stmt""#)]
+    r#type: &'static str,
 }
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, From, new)]
 pub struct ExprStmt {
     pub expr: Expr,
+    #[new(value = r#""expr_stmt""#)]
+    r#type: &'static str,
 }
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, new)]
 pub struct IfStmt {
+    pub cond: Expr,
     pub body: Block,
-    pub else_body: Block,
+    pub else_body: Option<Block>,
+    #[new(value = r#""if_stmt""#)]
+    r#type: &'static str,
 }
 
 #[derive(Debug, Eq, PartialEq, Serialize, Clone, new)]
 pub struct ForStmt {
-    pub init: Expr, // Expr, BinExpr that is = ? or a new DeclExpr?
-    pub condition: Expr,
+    pub init: Option<Expr>, // Expr, BinExpr that is = ? or a new DeclExpr?
```

Comment:
Could be either a DeclExpr or a BinExpr, considering that the loop counter could have been declared outside of this scope, or declared inline.